openedx/adg/lms/applications/forms.py

```diff
@@ -108,3 +108,36 @@ def __init__(self, *args, **kwargs):
         self.fields['phone_number'].required = True
         self.fields['city'].disabled = True
         self.fields['name'].disabled = True
+
+
+class UserApplicationCoverLetterForm(forms.ModelForm):
+    """
+    User Application Form for Cover Letter Page
+    """
+
+    class Meta:
+        model = UserApplication
+        fields = ['business_line', 'cover_letter_file', 'cover_letter']
+
+    def clean_cover_letter_file(self):
+        """
+        Validate cover letter file size is less than maximum allowed size
+        """
+        cover_letter_file = self.cleaned_data.get('cover_letter_file')
+        if cover_letter_file:
+            error = validate_file_size(cover_letter_file, RESUME_FILE_MAX_SIZE)
+            if error:
+                raise forms.ValidationError(error)
+        return cover_letter_file
+
+    def is_valid(self):
+        """
+        In addition to pre-defined validation it validates that either cover letter file or text is sent
+        """
+        valid = super(UserApplicationCoverLetterForm, self).is_valid()
+
+        if 'cover_letter_file' in self.changed_data and self.cleaned_data.get('cover_letter'):
```

Comment:
This if condition is a result of a whole lot of debugging, using other conditions does not produce the expected result.

we need to produce an error if both cover letter file and text are provided and that only happens when 'cover_letter_file' is in changed_data and 'cover_letter' has some text in cleaned_data.

The problem with using only cleaned_data is that even when the user removes file, cleaned_data hold the previously uploaded file instead of None in 'cover_letter_file'

The problem with using only changed_data is that even when the user removes file and adds text, changed_data holds both 'cover_letter' and 'cover_letter_file' keys (note: changed_data only hold keys not values so it can not be determined if they are empty or not)